pkg/metadata/provider_scaleway.go

```diff
@@ -210,7 +210,7 @@ func scalewayGetUserdata() ([]byte, error) {
 		return nil, errors.New("not able to found a free port below 1024")
 	}
 	defer conn.Close()
-	fmt.Fprintf(conn, "GET /user_data HTTP/1.0\r\n\r\n")
+	fmt.Fprintf(conn, "GET /user_data/cloud-init HTTP/1.0\r\n\r\n")
```

Comment:
As said in the coversation, it would be nice to support cloud-init + user data. One thing though, when looking at other providers, they only return the user data keys, and note the value. (Which would not be the case with cloud-init). @rn or @justincormack any thoughts on this? Shoud the `Extract` method returns only the user data keys or also the values?

Reply:
I looked around and I didn't see that you could add any other userdata than cloud-init stuff and that's why I did this and it's based on the following code https://git.launchpad.net/cloud-init/tree/cloudinit/sources/DataSourceScaleway.py?id=d4d11c78e5e78999356fd0c3d124b5a298735b65

Reply:
I do see that they also query vendor data which we could also do

Reply:
Ok, so I look into the user data stuff and from what I can tell you can provide any user data when creating an instance via the api, the question is then how do we merge this properly. From what I can tell most other providers in this package is just returning the userdata as is, and in most if not all cases that's just whatever the user provides and in most cases that's a cloud-config/cloud-init data.

k3os is depending on that kind of behaviour to just read text from userdata and if it's a script, add it to the run_cmd and if it's yaml use it as a cloud-config.